src/core/Navigation/ServiceNavigation/ServiceNavigation/ServiceNavigation.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { forwardRef, ReactNode, useState } from 'react';
+import {
+  HtmlButton,
+  HtmlDiv,
+  HtmlNav,
+  HtmlSpan,
+  HtmlUl,
+} from '../../../../reset';
+import styled from 'styled-components';
+import { SuomifiThemeConsumer, SuomifiThemeProp } from '../../../theme';
+import { baseStyles } from './ServiceNavigation.baseStyles';
+import classnames from 'classnames';
+import { Icon } from '../../../Icon/Icon';
+import { getConditionalAriaProp } from '../../../../utils/aria';
+
+export interface ServiceNavigationProps {
+  /** Use the `<ServiceNavigationItem>` components as children */
+  children: ReactNode;
+  /** Name for the navigation element. Don't use the word "navigation" since it will be read by screen reader regardless. */
+  'aria-label': string;
+  /** ID for the HTML nav element */
+  id?: string;
+  /**
+   * Normal or small screen variant
+   * @default normal
+   */
+  variant?: 'default' | 'smallScreen';
+  /**
+   * Initial expand status for the menu. Only applies to smallScreen variant
+   * @default true
+   */
+  initiallyExpanded?: boolean;
+  /** When using smallScreen variant, this is the text showed in the expander button */
+  smallScreenExpandButtonText?: string | ReactNode;
+  /** Custom classname to extend or customize */
+  className?: string;
+  /** Ref is forwarded to nav element. Alternative for React `ref` attribute. */
+  forwardedRef?: React.Ref<HTMLElement>;
+}
+
+const baseClassName = 'fi-service-navigation';
+const smallScreenClassName = `${baseClassName}--small-screen`;
+const listClassName = `${baseClassName}_list`;
+const smallScreenExpandButtonClassName = `${baseClassName}_expand-button`;
+
+const BaseServiceNavigation = ({
+  children,
+  'aria-label': ariaLabel,
+  id,
+  variant,
+  initiallyExpanded,
+  smallScreenExpandButtonText,
+  className,
+  forwardedRef,
+}: ServiceNavigationProps) => {
+  const initiallyExpandedValue =
+    initiallyExpanded !== undefined ? initiallyExpanded : true;
+  const [smallScreenNavOpen, setSmallScreenNavOpen] = useState(
+    initiallyExpandedValue,
+  );
+
+  return (
+    <HtmlDiv
+      className={classnames(baseClassName, className, {
+        [smallScreenClassName]: variant === 'smallScreen',
+      })}
+    >
+      {variant === 'smallScreen' && (
+        <HtmlButton
+          className={smallScreenExpandButtonClassName}
+          onClick={() => setSmallScreenNavOpen(!smallScreenNavOpen)}
+          aria-expanded={smallScreenNavOpen}
+        >
+          <HtmlSpan
```

Comment:
Should these styles come as class from baseStyles?

Reply:
Not sure what you mean here? Line 71 toggles between showing and hiding the nav, whereas line 72 renders the aria-expanded attribute based on whether the nav is opened or not

Reply:
Comment section shows this a bit strangely, but I meant rows 75-79 that have HtmlSpan component's styles inlined.

Reply:
Ah, I see. These were indeed meant to be moved to a CSS class but I forgot about it. Changed it now!